sdk/attestation/azure-security-attestation/src/main/java/com/azure/security/attestation/AttestationAsyncClient.java

```diff
@@ -8,43 +8,110 @@
 import com.azure.security.attestation.implementation.AttestationsImpl;
+import com.azure.security.attestation.implementation.MetadataConfigurationsImpl;
+import com.azure.security.attestation.implementation.SigningCertificatesImpl;
 import com.azure.security.attestation.models.AttestOpenEnclaveRequest;
 import com.azure.security.attestation.models.AttestSgxEnclaveRequest;
 import com.azure.security.attestation.models.AttestationResponse;
+import com.azure.security.attestation.models.AttestationSigner;
 import com.azure.security.attestation.models.CloudErrorException;
 import reactor.core.publisher.Mono;
 
 import java.nio.charset.StandardCharsets;
 import java.util.Objects;
 
+import static com.azure.core.util.FluxUtil.withContext;
+
 /** Initializes a new instance of the asynchronous AzureAttestationRestClient type. */
-@ServiceClient(builder = AttestationClientBuilder.class, isAsync = true)
+@ServiceClient(builder = AttestationAsyncClientBuilder.class, isAsync = true)
 public final class AttestationAsyncClient {
-    private final AttestationsImpl serviceClient;
+    private final AttestationsImpl attestImpl;
+    private final MetadataConfigurationsImpl metadataImpl;
+    private final SigningCertificatesImpl signerImpl;
 
     /**
      * Initializes an instance of Attestations client.
      *
-     * @param serviceClient the service client implementation.
+     * @param clientImpl the service client implementation.
+     */
+    AttestationAsyncClient(AttestationClientImpl clientImpl) {
+        this.attestImpl = clientImpl.getAttestations();
+        this.metadataImpl = clientImpl.getMetadataConfigurations();
+        this.signerImpl = clientImpl.getSigningCertificates();
+    }
+
+    /**
+     * Retrieves metadata about the attestation signing keys in use by the attestation service.
+     *
+     * @param context - Context for the operation
+     * @throws CloudErrorException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return any object.
+     */
+//    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<Object>> getOpenIdMetadataWithResponse(Context context) {
+        return this.metadataImpl.getWithResponseAsync(context);
+    }
+
+    /**
+     * Retrieves metadata about the attestation signing keys in use by the attestation service.
+     *
+     * @throws CloudErrorException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return any object.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Object> getOpenIdMetadata() {
+        // Forward the getOpenIdMetadata to the getOpenIdMetadataWithResponse API implementation.
+        return withContext(context -> this.getOpenIdMetadataWithResponse(context))
+            .map(Response::getValue);
```

Comment:
>.map(Response::getValue); [](http://example.com/codeflow?start=12&length=25)

Core has a utility method for this.
FluxUtil.toMono

So, you could use 
 return withContext(context -> this.getOpenIdMetadataWithResponse(context))
,flatMap(FluxUtil::toMono) instead.

Reply:
Is there a functional difference between the two? They seem equally terse, so I'm not 100% sure i understand why one is preferable to the other (except possibly that the flatMap explicitly returns a Mono<R> instead of implicitly converting R to Mono<R>

Reply:
If there is a possibility that response.getValue could be void - map won't work so please consider using this method instead.

Reply:
For these APIs, there really isn't, but I've switched everything that I can safely change.